packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -80,65 +99,46 @@ export class SummaryManager extends EventEmitter implements IDisposable {
             "SummaryManager",
             {all:{ clientId: () => this.latestClientId }});
 
-        this.connected = context.connected;
-        if (this.connected) {
-            this.setClientId(context.clientId);
-        }
+        this.connectedState.on("connected", this.handleConnected);
+        this.connectedState.on("disconnected", this.handleDisconnected);
+        this.latestClientId = this.connectedState.clientId;
 
         // Track ops until first (write) connect
         const opsUntilFirstConnectHandler = (clientId: string, details: ISequencedClient) => {
-            if (this.opsUntilFirstConnect === -1 && clientId === this.clientId) {
+            if (this.opsUntilFirstConnect === -1 && clientId === this.connectedState.clientId) {
                 context.quorum.off("addMember", opsUntilFirstConnectHandler);
                 this.opsUntilFirstConnect = details.sequenceNumber - this.context.deltaManager.initialSequenceNumber;
             }
         };
         context.quorum.on("addMember", opsUntilFirstConnectHandler);
 
-        clientElection.on("electedSummarizerChanged", () => this.refreshSummarizer());
-
         this.initialDelayTimer = new PromiseTimer(initialDelayMs, () => { });
         this.initialDelayP = this.initialDelayTimer?.start() ?? Promise.resolve();
-
-        this.refreshSummarizer();
-    }
-
-    public setConnected(clientId: string) {
-        this.updateConnected(true, clientId);
-    }
-
-    public setDisconnected() {
-        this.updateConnected(false);
-    }
-
-    private setClientId(clientId: string | undefined): void {
-        this.clientId = clientId;
-        if (clientId !== undefined) {
-            this.latestClientId = clientId;
-            if (this.runningSummarizer !== undefined) {
-                this.runningSummarizer.updateOnBehalfOf(clientId);
-            }
-        }
     }
 
-    public on(event: "summarizer", listener: (clientId: string) => void): this;
-    public on(event: string, listener: (...args: any[]) => void): this {
-        return super.on(event, listener);
+    /**
+     * Until start is called, the SummaryManager won't begin attempting to start summarization.  This ensures there's
+     * a window between construction and starting where the caller can attach listeners.
+     */
+    public start(): void {
```

Comment:
I don't like start.. couldn't there be a callback passed to the constructor?

Something like?
```ts
constructor(eventAttacher: (summaryManager: IEventProvider<ISummaryManagerEvents>) => void) {
    eventAttacher(this);
}
```

Or more generic?
```ts
constructor(setup: (summaryManager: SummaryManager) => void) {
    setup(this);
}
```

Definitely wouldn't hold the PR for this though.

Reply:
Returning a stationary object from the constructor and letting the caller hook up before the action starts is easier to use (otherwise you end up with weird patterns like `IDocumentDeltaConnection.initialMessages` and `Container.load()`).  Maybe prefer a different name, `enable()` or similar?

Reply:
I just don't like that it can be called multiple times with weird behavior, but it's fine.. pretty common pattern.